library/core/src/slice/cmp.rs

```diff
@@ -13,7 +13,9 @@ extern "C" {
     ///
     /// Returns 0 for equal, < 0 for less than and > 0 for greater
     /// than.
-    // FIXME(#32610): Return type should be c_int
+    #[cfg(any(target_arch = "avr", target_arch = "msp430",))]
+    fn memcmp(s1: *const u8, s2: *const u8, n: usize) -> i16;
+    #[cfg(not(any(target_arch = "avr", target_arch = "msp430",)))]
     fn memcmp(s1: *const u8, s2: *const u8, n: usize) -> i32;
```

Comment:
We actually have `c_int` in libcore now - https://doc.rust-lang.org/nightly/core/ffi/type.c_int.html
Although it's not defined correctly for AVR and MSP430.

So the correct definition of `c_int` can go there now, and `memcmp` can simply use it.

Reply:
Neat!  Thanks